doc/build-helpers/images/dockertools.section.md

```diff
@@ -36,84 +48,216 @@ buildImage {
+
+  :::{.caution}
+  Using this attribute requires the `kvm` device to be available.
+  :::
+
+  :::{.note}
+  At the time the script in `runAsRoot` is run, the files specified directly in `copyToRoot` will be present in the VM, but their dependencies might not be there yet.
+  Copying their dependencies into the generated image is a step that happens after `runAsRoot` finishes running.
+  :::
+
+    _Default value:_ `null`.
+
+`extraCommands` _optional_
+
+: A bash script that will run before the layer created by `buildImage` is finalised.
+  This is similar to `runAsRoot`, but the script specified in `extraCommands` is **not** run as root, and does not involve creating a VM.
+  It is simply run as part of building the derivation that outputs the layer created by `buildImage`.
+
+    _Default value:_ `""`.
+
+`config` _optional_
+
+: Used to specify the configuration of the containers that will be started off the generated image.
+  Must be an attribute set, with each attribute as listed in the [Docker Image Specification v1.3.0](https://github.com/moby/moby/blob/46f7ab808b9504d735d600e259ca0723f76fb164/image/spec/spec.md#image-json-field-descriptions).
+
+    _Default value:_ `null`.
+
+`architecture` _optional_
+
+: Used to specify the image architecture.
+  This is useful for multi-architecture builds that don't need cross compiling.
+  If specified, its value should follow the [OCI Image Configuration Specification](https://github.com/opencontainers/image-spec/blob/main/config.md#properties), which should still be compatible with Docker.
+
+    _Default value:_ the same value from `pkgs.go.GOARCH`.
+
+`diskSize` _optional_
+
+: Controls the disk size (in megabytes) of the VM used to run the script specified in `runAsRoot`.
+  This attribute is ignored if `runAsRoot` is `null`.
+
+    _Default value:_ 1024.
+
+`buildVMMemorySize` _optional_
+
+: Controls the amount of memory (in megabytes) provisioend for the VM used to run the script specified in `runAsRoot`.
+  This attribute is ignored if `runAsRoot` is `null`.
+
+    _Default value:_ 512.
+
+`created` _optional_
+
+: Specifies the time of creation of the generated image.
+  This should be either an ISO-8601-formatted date and time or `"now"`, in which case `buildImage` will use the current date.
+
+  :::{.caution}
+  Using `"now"` means that the generated image will not be reproducible anymore (because the date will always change whenever it's built).
+  :::
+
+    _Default value:_ `"1970-01-01T00:00:01Z"`.
+
+`uid` _optional_
+
+: The uid of the user that will own the files packed in the new layer built by `buildImage`.
+
+    _Default value:_ 0.
+
+`gid` _optional_
+
+: The gid of the group that will own the files packed in the new layer built by `buildImage`.
+
+    _Default value:_ 0.
+
+`contents` **DEPRECATED**
+
+: This attribute is deprecated, and users are encouraged to use `copyToRoot` instead.
+
+`buildImage` defines a few [`passthru`](#var-stdenv-passthru) attributes:
```

Comment:
It would be good to have a cleaner separation from inputs vs "outputs", maybe a nested heading for each would work.